tools/Google.Cloud.Tools.ConstructorGenerator/Program.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.Tools.Common;
+using Grpc.Core;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Loader;
+
+namespace Google.Cloud.Tools.ConstructorGenerator
+{
+    /// <summary>
+    /// One-off tool to generate constructors in partial classes for gRPC compatibility.
+    /// </summary>
+    class Program
+    {
+        private const string TestTargetFramework = "netcoreapp2.1";
+
+        static void Main(string[] args)
+        {
+            var catalog = ApiCatalog.Load();
+
+            var generateAwkward = args.Length == 1 && args[0] == "--awkward";
+
+            var apis = catalog.Apis
+                // Only use gRPC APIs
+                .Where(api => api.Type == ApiType.Grpc)
+                // If the minor version is greater than 0, there must be a GA release.
```

Comment:
What about the ones that are GA on a zero minor? Like 3.0.0 GA? If we don't have any like those, this is fine being a one off.

Reply:
3.0.0 GA is fine - api.IsReleaseVersion will be true. The idea is to generate constructors for "2.1.0-beta01" but not for "2.0.0-beta01".